packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -321,6 +323,10 @@ export default function useBlockDropZone( {
 		stopDragging,
 	} = unlock( useDispatch( blockEditorStore ) );
 
+	const { sectionRootClientId } = unlock( getSettings() );
+
+	const isZoomOutMode = __unstableGetEditorMode() === 'zoom-out';
```

Comment:
Since these are grabbing values, I was wondering if these should either be:

* Retrieved in the traditional way of a `useSelect` call that returns an object that we then destructure to assign values, or:
* Move the assignment of these values to the callback within the throttled function below?

The `useBlockDropZone` hook is a little unusual in that we get references to a bunch of functions that are later called in the throttled function, so I was wondering if it's good to try to keep to the method calls to happen either in a `useSelect` callback, or within the throttled callback, but not at the root of the hook. I think the main thing to look for here is whether we're currently calling `getSettings()` and `__unstableGetEditorMode()` more frequently than we need to 🤔

Reply:
Yes I agree thanks for flagging. Performance was my next concern if folks felt this was on track.

I've updated so that the comparisons are only called within the throttled callback. The callback dependencies have increased by x2 but one is a stable function reference and the other relates to whether or not we are in zoom out mode.